src/Messaging/src/RabbitMQ/Extensions/RabbitServicesExtensions.cs

```diff
@@ -244,11 +244,18 @@ public static IServiceCollection ConfigureRabbitOptions(this IServiceCollection
                 .Bind(config.GetSection(RabbitOptions.PREFIX))
                 .Configure<IServiceProvider>((options, provider) =>
                 {
-                    var connectionFactory = provider.GetService<RC.IConnectionFactory>() as RC.ConnectionFactory;
-
-                    if (connectionFactory is not null)
+                    using (var scope = provider.CreateScope())
                     {
-                        options.Addresses = $"{connectionFactory.HostName}:{connectionFactory.Port}";
+                        var connectionFactory = scope.ServiceProvider.GetService<RC.IConnectionFactory>() as RC.ConnectionFactory;
+
+                        if (connectionFactory is not null)
+                        {
+                            options.Addresses = $"{connectionFactory.UserName}:{connectionFactory.Password}@{connectionFactory.HostName}:{connectionFactory.Port}";
+                            options.VirtualHost = connectionFactory.VirtualHost;
+                            options.Host = connectionFactory.HostName;
+                            options.Username = connectionFactory.UserName;
+                            options.Password = connectionFactory.Password;
+                        }
                     }
```

Comment:
It is somewhat accurate - but we have a test that passes in windows , not linux on build server